### PR TITLE
Update Min Hold Time

### PR DESCRIPTION
**Pull request by TJ-Everett**

### CODE REVIEW

File: src/client/token/handler/diamond/ERC721NonTaggedRuleFacet.sol [hunk 1]
```diff
@@ -17,6 +17,8 @@ import "src/client/token/handler/ruleContracts/HandlerTokenMinTxSize.sol";
 import "src/client/token/handler/ruleContracts/HandlerTokenMinHoldTime.sol";
 import "src/client/token/handler/ruleContracts/HandlerTokenMaxDailyTrades.sol";
 
+import "forge-std/console.sol";
```
Comment (VoR0220) [inline src/client/token/handler/diamond/ERC721NonTaggedRuleFacet.sol]: We need to drop this

File: test/client/token/ERC721/util/ERC721Util.sol [hunk 2]
```diff
@@ -125,15 +125,22 @@ abstract contract ERC721Util is TokenUtils, DummyNFTAMM {
         ERC721NonTaggedRuleFacet(address(assetHandler)).setTokenMinTxSizeIdFull(actions, ruleIds);
     }
 
+    //TODO Remove this function once tests are complete
```
Comment (VoR0220) [inline test/client/token/ERC721/util/ERC721Util.sol]: So...uhhh...we removing this?

**Comment by VoR0220** [comment]
Very very minute things to change. This looks very gtg assuming tests all pass.

File: src/client/application/AppManager.sol [hunk 3]
```diff
@@ -21,6 +21,7 @@ import "src/client/token/HandlerTypeEnum.sol";
 import {IApplicationEvents, IIntegrationEvents} from "src/common/IEvents.sol";
 import {ActionTypes} from "src/common/ActionEnum.sol";
 
+import "forge-std/console.sol";
```
Comment (ShaneDuncan602) [inline src/client/application/AppManager.sol]: I think this got left behind.

File: src/client/token/handler/diamond/ERC721NonTaggedRuleFacet.sol [hunk 4]
```diff
@@ -258,7 +259,7 @@ contract ERC721NonTaggedRuleFacet is
         }
 
         if (minHoldTime.tokenMinHoldTime[potentialOppositeAction].active && minHoldTime.ownershipStart[_tokenId] > 0) {
-            IRuleProcessor(handlerBase).checkTokenMinHoldTime(minHoldTime.tokenMinHoldTime[potentialOppositeAction].period, minHoldTime.ownershipStart[_tokenId]);
+            IRuleProcessor(handlerBase).checkTokenMinHoldTime(minHoldTime.tokenMinHoldTime[potentialOppositeAction].ruleId, minHoldTime.ownershipStart[_tokenId]);
         }
```
Comment (ShaneDuncan602) [inline src/client/token/handler/diamond/ERC721NonTaggedRuleFacet.sol:261]: This is no longer a simple rule and should probably not be in this function any more. I think it should either go into nontagged rules check or we change the name of the function to _checkMinHoldTime.

File: src/client/token/handler/ruleContracts/HandlerTokenMinHoldTime.sol [hunk 5]
```diff
@@ -98,13 +98,13 @@ contract HandlerTokenMinHoldTime is RuleAdministratorOnly, ITokenHandlerEvents,
      * @dev Set the TokenMinHoldTime.
      * @notice that setting a rule will automatically activate it.
      * @param _action the action type to set the rule
-     * @param _minHoldTimeHours the min hold time in hours
+     * @param _ruleId the rule id
      */
-    function setTokenMinHoldTimeIdUpdate(ActionTypes _action, uint32 _minHoldTimeHours) internal {
-        if (_minHoldTimeHours == 0) revert ZeroValueNotPermited();
-        if (_minHoldTimeHours > MAX_HOLD_TIME_HOURS) revert PeriodExceeds5Years();
+    // slither-disable-next-line calls-loop
```
Comment (ShaneDuncan602) [inline src/client/token/handler/ruleContracts/HandlerTokenMinHoldTime.sol:103]: Is this needed here?

File: src/protocol/economic/ruleProcessor/RuleDataFacet.sol [hunk 6]
```diff
@@ -239,4 +239,17 @@ contract RuleDataFacet is Context, RuleAdministratorOnly, IEconomicEvents, IInpu
...
+        if (_minHoldtime > 43830) revert PeriodExceeds5Years();
+        RuleS.TokenMinHoldTimeS storage data = Storage.tokenMinHoldTimeStorage();
+        NonTaggedRules.TokenMinHoldTime memory rule = NonTaggedRules.TokenMinHoldTime(_minHoldtime);
+        uint32 ruleId = data.tokenMinHoldTimeIndex;
+        data.tokenMinHoldTimeRules[ruleId] = rule;
+        emit AD1467_ProtocolRuleCreated(TOKEN_MIN_HOLD_TIME, ruleId, new bytes32[](0));
+        ++data.tokenMinHoldTimeIndex;
+        return ruleId;
+    }
+
```
Comment (ShaneDuncan602) [inline src/protocol/economic/ruleProcessor/RuleDataFacet.sol:260]: Missing natspec comments

File: src/protocol/economic/ruleProcessor/RuleStoragePositionLib.sol [hunk 7]
```diff
@@ -150,6 +151,13 @@ library RuleStoragePositionLib {
         }
     }
 
+    function tokenMinHoldTimeStorage() internal pure returns (IRuleStorage.TokenMinHoldTimeS storage ds) {
+        bytes32 position = TOKEN_MIN_HOLD_TIME_POSITION;
+        assembly {
+            ds.slot := position
+        }
+    }
+
```
Comment (ShaneDuncan602) [inline src/protocol/economic/ruleProcessor/RuleStoragePositionLib.sol:164]: Missing natspec comments

**Review by ShaneDuncan602 — CHANGES_REQUESTED** [review]
Really fast turnaround on this. Great job. I just found a few small things.

**Review by Palmerg4 — COMMENTED** [review]
Looks great, same suggestions as the other guys. Nice work

File: src/client/token/handler/ruleContracts/HandlerTokenMinHoldTime.sol (hunk 5 above)
Comment (TJ-Everett) [reply src/client/token/handler/ruleContracts/HandlerTokenMinHoldTime.sol:103]: I think so, the validate function right below it calls a looped function in the createArray function.

**Review by ShaneDuncan602 — APPROVED** [review]
You rocked it!

**Review by VoR0220 — APPROVED** [review]

**Review by Palmerg4 — APPROVED** [review]

**Pull request merged**